doc/source/security/encrypt-traffic.rst

```diff
@@ -0,0 +1,446 @@
+This includes traffic between client applications and cluster nodes,
+:term:`SST`, :term:`IST`, write-set replication, and various service messages.
+
+.. contents::
+   :local:
+   :depth: 1
+
+.. _generate-keys-certs:
+
+Generating Keys and Certificates
+================================
+
+To encrypt |PXC| traffic, you must generate the following sets of files:
+
+* *Certificate Authority (CA) key and certificate*
+  to sign the server and client certificates.
+
+* *Server key and certificate*
+  to secure database server activity and write-set replication traffic.
+
+* *Client key and certificate*
+  to secure client communication traffic.
+
+These files should be generated using `OpenSSL <https://www.openssl.org/>`_.
+
+.. note:: The ``Common Name`` value
+   used for the server and client keys and certificates
+   must differ from that value used for the CA certificate.
+
+.. _generate-ca-key-cert:
+
+Generating CA Key and Certificate
+---------------------------------
+
+The Certificate Authority is used to verify the signature on certificates.
+
+1. Generate the CA key file::
+
+    $ openssl genrsa 2048 > ca-key.pem
+
+#. Generate the CA certificate file::
+
+    $ openssl req -new -x509 -nodes -days 3600
+        -key ca-key.pem -out ca.pem
+
+.. _generate-server-key-cert:
+
+Generating Server Key and Certificate
+-------------------------------------
+
+1. Generate the server key file::
+
+    $ openssl req -newkey rsa:2048 -days 3600 \
+        -nodes -keyout server-key.pem -out server-req.pem
+
+#. Remove the passphrase::
+
+    $ openssl rsa -in server-key.pem -out server-key.pem
+
+#. Generate the server certificate file::
+
+    $ openssl x509 -req -in server-req.pem -days 3600 \
+        -CA ca.pem -CAkey ca-key.pem -set_serial 01 \
+        -out server-cert.pem
+
+.. _generate-client-key-cert:
+
+Generating Client Key and Certificate
+-------------------------------------
+
+1. Generate the client key file::
+
+    $ openssl req -newkey rsa:2048 -days 3600 \
+        -nodes -keyout client-key.pem -out client-req.pem
+
+#. Remove the passphrase::
+
+    $ openssl rsa -in client-key.pem -out client-key.pem
+
+#. Generate the client certificate file::
+
+    $ openssl x509 -req -in client-req.pem -days 3600 \
+        -CA ca.pem -CAkey ca-key.pem -set_serial 01 \
+        -out client-cert.pem
+
+.. _verify-certs:
+
+Verifying Certificates
+----------------------
+
+To verify that the server and client certificates
+are correctly signed by the CA certificate,
+run the following command::
+
+ $ openssl verify -CAfile ca.pem server-cert.pem client-cert.pem
+
+If the verification is successful, you should see the following output::
+
+ server-cert.pem: OK
+ client-cert.pem: OK
+
+Diploying Keys and Certificates
+-------------------------------
+
+Use a secure method (for example, ``scp`` or ``sftp``)
+to send the key and certificate files to each node.
+Place them under the :file:`/etc/mysql/certs/` directory
```

Comment:
We should mention the permissions that should be on this directory, most likely read-only by mysql, no other access, so probably x400